molo/forms/tests/test_reaction_questions.py

```diff
@@ -0,0 +1,528 @@
+            self.section_index, title='Your mind')
+
+        self.yourmind_fr = self.mk_section_translation(
+            self.yourmind, self.french, title='Your mind in french')
+
+        self.mk_main2()
+        self.language_setting2 = Languages.objects.create(
+            site_id=self.main2.get_site().pk)
+        self.english2 = SiteLanguageRelation.objects.create(
+            language_setting=self.language_setting2,
+            locale='en',
+            is_active=True)
+
+        self.spanish = SiteLanguageRelation.objects.create(
+            language_setting=self.language_setting2,
+            locale='es',
+            is_active=True)
+
+        self.yourmind2 = self.mk_section(
+            self.section_index2, title='Your mind2')
+
+        self.reaction_index = self.get_reaction_index(self.main)
+        self.reaction_index2 = self.get_reaction_index(self.main2)
+
+    def test_can_react_on_article(self):
+        article = self.mk_article(self.yourmind)
+        article.save_revision().publish()
+        question = self.mk_reaction_question(self.reaction_index, article)
+        self.user = self.login()
+
+        response = self.client.get(article.url)
+
+        self.assertContains(response, question.title)
+
+        for choice in question.get_children():
+            self.assertContains(response, choice.title)
+
+        choice1 = question.get_children().first()
+        self.assertEqual(ReactionQuestionResponse.objects.all().count(), 0)
+
+        kw = {'question_id': question.id, 'article_slug': article.slug}
+        data = {'choice': choice1.id}
+        response = self.client.post(
+            reverse('molo.forms:reaction-vote', kwargs=kw), data=data)
+
+        self.assertEqual(ReactionQuestionResponse.objects.all().count(), 1)
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(
+            response['Location'],
+            '/forms/reaction/test-page-0/{}/yes/feedback/'.format(question.pk)
+        )
+
+        response = self.client.get(
+            '/forms/reaction/test-page-0/{}/yes/feedback/'.format(question.pk)
+        )
+
+        self.assertContains(
+            response, '<a href="/sections-main-1/your-mind/test-page-0/">')
+
+        self.assertContains(response, 'well done')
+
+        # test user can only submit once
+        response = self.client.post(
+            reverse('molo.forms:reaction-vote', kwargs=kw),
+            data=data)
+        self.assertEqual(ReactionQuestionResponse.objects.all().count(), 1)
+        self.assertEqual(
+            ReactionQuestionResponse.objects.last().choice.pk, choice1.pk)
+
+        messages = list(get_messages(response.wsgi_request))
+        self.assertEqual(len(messages), 1)
+        # this should show if the submit is not done via ajax
+        self.assertIn(
+            'You have already given feedback on this article.',
+            str(messages[0])
+        )
+
+    def test_can_react_on_article_ajax_call(self):
```

Comment:
I think this test needs to submit the reaction question response twice (like the one above).
If I'm not mistaken there should be 0 ReactionQuestionResponse objects at the start of this test. The first post to submit a response creates one and then the second post should change the chosen value. Currently you're only creating the first one. So you wouldn't get an error message anyway and you're not testing that the value gets changed on the second submission.